Gems/ROS2/Code/Source/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp

```diff
@@ -190,7 +190,8 @@ namespace ROS2
         };
 
         // Use the URDF/SDF file name stem the prefab name
-        AZStd::string prefabName = AZStd::string(AZ::IO::PathView(filePath).Stem().Native());
+        auto fileStem = AZ::IO::PathView(filePath).Stem();
+        AZStd::string prefabName = AZStd::string::format("%.*s.prefab", AZ_PATH_ARG(fileStem));
```

Comment:
Suggestion: As an alternative, you could do...
```
AZ::IO::Path prefabName = filePath;
prefabName.ReplaceExtension(".prefab");
```

Not sure whether that or string::format would run faster, I've got 0 objections with leaving it the way it is too.

Reply:
The logic would still have to call `.File()` has the filePath is the full path containing any parent directories.
Technically it could be something such as
```
AZ::IO::Path prefabName = AZ::IO::Path(filePath).Filename();
prefabName.ReplaceExtension(".prefab");
```

Now the AZ::IO::Path call would construct a temporary AZStd::string under the hood and then create another AZStd::string that contains the full file name, such as  `/home/user/URDF/ur.urdf.xacro` -> `ur.urdf.xacro`.
The replace extension call would update the string in place which probably will not need to allocate if the string is <22 characters or the allocated memory can fit an additional character due to replacing "xacro" with "prefab"